src/fidgets/ui/LinksInput.tsx

```diff
@@ -0,0 +1,282 @@
+import { FaPencil } from "react-icons/fa6";
+import CSSInput from "@/common/components/molecules/CSSInput";
+
+export interface LinksInputProps {
+  value: Link[];
+  onChange?: (value: Link[]) => void;
+}
+
+const Box = styled.div`
+  width: 100%;
+`;
+
+const TextFieldRoot = styled.div<{ isActive: boolean }>`
+  display: flex;
+  align-items: center;
+  position: relative;
+  background: white;
+  border: ${({ isActive }) =>
+    isActive ? "2px solid lightblue" : "1px solid lightgray"};
+  padding: 5px;
+  border-radius: 4px;
+  width: 100%;
+`;
+
+const TextFieldSlot = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const TextFieldInput = styled.input`
+  border: none;
+  outline: none;
+  flex: 1;
+  padding: 0 5px;
+`;
+
+const LinkContainer = styled.div`
+  margin-bottom: 1em;
+`;
+
+const Title = styled.p`
+  font-size: 0.875rem;
+  color: black;
+  font-weight: bold;
+`;
+
+const SubTitle = styled.p`
+  font-size: 0.875rem;
+  color: gray;
+`;
+
+const AddLinkButton = styled.button`
+  display: flex;
+  align-items: center;
+  margin-top: 10px;
+  background: none;
+  border: none;
+  color: #007bff;
+  cursor: pointer;
+  font-size: 1rem;
+
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
+const LinksInput = forwardRef<HTMLInputElement, LinksInputProps>(
+  (props, ref) => {
+    const { value = [], onChange } = props;
+
+    const [linkStates, setLinkStates] = useState<
+      {
+        isAvatarInputDisplayed: boolean;
+        confirmed: boolean;
+        isActive: boolean;
+      }[]
+    >([]);
+
+    useEffect(() => {
+      if (value.length === 0) {
+        const defaultLink = {
+          text: "My Awesome Link",
+          url: "https://nounspace.com",
+          avatar:
+            "https://nounspace.mypinata.cloud/ipfs/QmSkE84i5LoA6b6VZBZe4Ku4MTo5ueYhCc3Hh7EY8JM3pb",
+          description: "Best dApp ever",
+        };
+        onChange?.([defaultLink]);
+        setLinkStates([
+          { isAvatarInputDisplayed: false, confirmed: false, isActive: false },
+        ]);
+      } else if (linkStates.length !== value.length) {
+        setLinkStates(
+          value.map(() => ({
+            isAvatarInputDisplayed: false,
+            confirmed: false,
+            isActive: false,
+          })),
+        );
+      }
+    }, [value, onChange, linkStates.length]);
+
+    const handleLinkChange = useCallback(
+      (index: number, newLink: Link) => {
+        const updatedLinks = [...value];
+        updatedLinks[index] = newLink;
+        onChange?.(updatedLinks);
+
+        const updatedLinkStates = [...linkStates];
+        if (newLink.url !== "") {
+          updatedLinkStates[index].isAvatarInputDisplayed = true;
+        } else {
+          updatedLinkStates[index].isAvatarInputDisplayed = false;
+        }
+        setLinkStates(updatedLinkStates);
+      },
+      [value, onChange, linkStates],
+    );
+
+    const addNewLink = () => {
+      onChange?.([
+        ...value,
+        {
```

Comment:
Two things:
1. Does this support saving with any of these fields empty? If so, we should just null these values and use them as the placeholder value in the display
2. These values are duplicated, please move it into a variable that is referenced instead of hard coding it twice

Reply:
1. it does, what happens right now is that i have an Logo, text and url placeholder. (there is a strange behavior which is the first placeholder example only appears when the user clicks on settings the first time. 

2. will do. checking today